android/clientlib/src/main/java/com/solana/mobilewalletadapter/clientlib/protocol/MobileWalletAdapterClient.java

```diff
@@ -145,13 +146,17 @@ public static class AuthorizationResult {
         public final Uri walletUriBase;
         @NonNull @Size(min = 1)
         public final AuthorizedAccount[] accounts;
+        @Nullable
+        public final SignInResult signInResult;
```

Comment:
The ergonomics of this object feels...odd, given its overlap with some of the values in the `AuthorizedAccount` object. But I don't really think there is a better option than this. I think you were forced to bolt this on.

Reply:
yeah im not thrilled with it. I could subclass the authorize objects but that would add complexity elsewhere so this felt the least bad (lol).